relayer-cli/src/commands/tx/client.rs

```diff
@@ -240,24 +248,32 @@ impl Runnable for TxUpgradeClientCmd {
         let client = ForeignClient::find(src_chain, dst_chain, &self.client_id)
             .unwrap_or_else(exit_with_unrecoverable_error);
 
-        // Assumption: this query is run while the chain is halted as a result of an upgrade
-        let src_upgrade_height = {
-            let src_application_height = match client.src_chain().query_latest_height() {
+        // In order to perform the client upgrade, the chain is paused at the height specified by
+        // the user. When the chain is paused, the application height reports a height of 1 less
+        // than the height according to Tendermint. As a result, the target height at which the
+        // upgrade occurs at (the application height) is 1 less than the height specified by
+        // the user.
+        let target_application_upgrade_height = match self.target_upgrade_height.decrement() {
+            Ok(height) => height,
+            Err(e) => Output::error(format!("{}", e)).exit(),
+        };
+
+        let mut src_application_latest_height = match client.src_chain().query_latest_height() {
+            Ok(height) => height,
+            Err(e) => Output::error(format!("{}", e)).exit(),
+        };
+
+        // Wait until the client's application height reaches the target application upgrade height
+        while src_application_latest_height < target_application_upgrade_height {
+            thread::sleep(Duration::from_millis(200));
```

Comment:
This seems a bit low to me given that blocks typically take >1 seconds to be emitted. Let's use a 1 second interval perhaps? What do you think @seanchen1991 @adizere?

Reply:
Right, probably does not matter too much indeed, but let's go with 500ms still?